node.go

```diff
@@ -477,17 +506,15 @@ func (n *Node) handleVoteRequest(vreq *pb.VoteRequest) bool {
 
 	n.setVote(vreq.Candidate)
 
-	// Write our state if needed.
-	if saveState {
-		if err := n.writeState(); err != nil {
-			// We have failed to update our state. Process the error
-			// and deny the vote.
-			n.handleError(err)
-			n.setVote(NO_VOTE)
-			n.rpc.SendVoteResponse(vreq.Candidate, deny)
-			n.resetElectionTimeout()
-			return true
-		}
+	// Write our state.
+	if err := n.writeState(); err != nil {
```

Comment:
This is writing state needed for leadership election, so seems like it should be graft's responsibility?